department-of-veterans-affairs/caseflow
# Mkerns/caseflow 2241 customized on hold modal

Resolves #{github issue number}

### Description
I added a conditional rendering SelectBox for the customHold options and added the error handling functionality when a user tries to input a hold longer than 30 days.

### Acceptance Criteria
- [ ] Options in dropdown only increase up to 30 days
- [ ] When selecting "Custom" in the dropdown, If user attempts to enter a period of more than 30 days we should not allow the form to submit and show an error message saying "The maximum allowable on hold time is 30 days"
- [ ] User can enter any other length of hold time between 1 - 30 days

### Testing Plan
1. Go to PO/VISN Queue
2. Select appeal
3. Click on Available Actions dropdown
4. Select on hold
5. Select hold options dropdown
6. Note that you should only see available options for 15 Days, 30 Days, and Custom.
7. Select Custom
8. New custom hold text field should appear
9. Try entering a hold longer than 30 days
10. Add instructions
11. Click Submit
12. Should trigger error for custom hold saying "The maximum allowable on hold time is 30 days"
13. Enter a valid hold time
14. Resubmit and the task should be set on hold for the allowed set time

https://user-images.githubusercontent.com/69862429/141200564-b6980996-02ee-4970-b74c-5a5898dc03dc.mov
